src/navigation/Navigation.js

```diff
@@ -0,0 +1,123 @@
+import React, { useContext, useEffect } from "react";
+import {
+  NavigationContainer,
+  DefaultTheme,
+  DarkTheme,
+} from "@react-navigation/native";
+import Icon from "react-native-vector-icons/MaterialCommunityIcons";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import { createBottomTabNavigator } from "@react-navigation/bottom-tabs";
+import { useTranslation } from "react-i18next";
+
+import HomeStack from "./stacks/HomeStack";
+import SettingsScreen from "../pages/SettingsScreen";
+import routes from "./routes";
+import { ThemeContext } from "../context/ThemeContext/ThemeProvider";
+import BookmarkStack from "./stacks/BookmarkStack/BookmarkStack";
+import SearchStack from "./stacks/SearchStack/SearchStack";
+import theme from "../styles/theme/theme";
+import { SET_BOOKMARKS } from "../context/ThemeContext/types";
+
+export default function Navigation() {
+  const Tab = createBottomTabNavigator();
+  const { state, dispatch } = useContext(ThemeContext);
+  const { t, i18n } = useTranslation();
+
+  useEffect(() => {
+    getData("theme").then((data) => {
+      if (data) {
+        dispatch({ type: data });
+      }
+    });
+    getData("lang").then((data) => {
+      if (data) {
+        i18n.changeLanguage(data);
+      }
+    });
+    getData("bookmarks").then((data) => {
+      if (data) {
+        dispatch({
+          type: SET_BOOKMARKS,
+          payload: JSON.parse(data),
+        });
+      }
+    });
+  }, []);
+  const getData = async (key) => {
```

Comment:
Veri nereden getiriliyor? Biraz daha açıklayıcı bir isim tercih edilmeli.
```suggestion
  const getDataFromAsyncStore = async (key) => {
```